### PR TITLE
docs: Update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by rakechill**
**Description**
Update CONTRIBUTING.md with common FAQs + an explanation on testing. 

Still TODO:
- [ ] I'm not sure how best to explain when to use `pkg/test/environment.go` vs. `test/common/environment.go`. Will reach out to the team to help with this one.
- [ ] I can move our developer guide here once Bryce's README updates are merged.

**Does this change impact docs?**
- [X] Yes, PR includes docs updates
- [ ] Yes, issue opened: # 
- [ ] No

### CODE REVIEW

File: CONTRIBUTING.md
Comment (Bryce-Soghigian) [inline CONTRIBUTING.md:46]: I don't think mentioning all of these is the right idea, ideally we should just remove the klog instances all together.

File: CONTRIBUTING.md
Comment (Bryce-Soghigian) [inline CONTRIBUTING.md:9]: This should probably also include the getting started dev environment for codespaces, imo this document is the right place for that :)

File: CONTRIBUTING.md
Comment (Bryce-Soghigian) [inline CONTRIBUTING.md:11]: Do we want to also mention coveralls and acceptance criteria for tests? I do like the breakdown between Unit using go test, then ginkgo for behavioral testing.

File: CONTRIBUTING.md
Comment (Bryce-Soghigian) [reply CONTRIBUTING.md:46]: Also note https://github.com/kubernetes-sigs/karpenter/issues/922

Comment (rakechill) [reply CONTRIBUTING.md:46]: I think it's worthwhile to reflect the current state of our repo. These can be updated to reflect how we want to use logging in the future. What's the concern with stating this here?

From your comments here, maybe we can also create two issues:
- a related issue for using controller runtime instead of knative
- remove klog invocations + replace with ???

File: CONTRIBUTING.md
Comment (rakechill) [reply CONTRIBUTING.md:9]: Yep I referenced that in the PR description. Just waiting on your docs PR to be merged first.

File: CONTRIBUTING.md
Comment (rakechill) [reply CONTRIBUTING.md:11]: Probably worth mentioning coveralls. What do you mean by acceptance criteria?

File: CONTRIBUTING.md
Comment (Bryce-Soghigian) [reply CONTRIBUTING.md:46]: Res that would be fantastic!

File: CONTRIBUTING.md
Comment (Bryce-Soghigian) [reply CONTRIBUTING.md:9]: gotcha!

File: CONTRIBUTING.md
Comment (Bryce-Soghigian) [reply CONTRIBUTING.md:11]: i was thinking code coverage for all new functions, but maybe coveralls does enough

**Review by Bryce-Soghigian — DISMISSED** [review]
Approved after readme change goes in

File: CONTRIBUTING.md
Comment (tallaxes) [reply CONTRIBUTING.md:9]: Longer term there should be a separate developer's guide (can be referenced from CONTRIBUTING.md), but for now it would be ok to just include it here.

File: CONTRIBUTING.md
Comment (tallaxes) [inline CONTRIBUTING.md:12]:
```suggestion
> A [GitHub Codespaces]((https://github.com/features/codespaces)) development flow is described below, which you can use to test Karpenter functionality on your own cluster, and to aid rapid development of this project. 
```

File: CONTRIBUTING.md
Comment (tallaxes) [inline CONTRIBUTING.md:38]: Is this sitll a thing?

File: CONTRIBUTING.md
Comment (tallaxes) [inline CONTRIBUTING.md:83]:
```suggestion
* `settings.go` is in the [apis](https://github.com/Azure/karpenter/blob/main/pkg/apis/settings/settings.go) package and provides settings needed for Karpenter to access a particular cluster.
```

**Review by tallaxes — APPROVED** [review]

File: CONTRIBUTING.md
Comment (rakechill) [reply CONTRIBUTING.md:38]: I didn't observe this when I just created a new one, so I'll remove for now.

**Pull request merged**
